src/puffin/src/puffin_manager/cached_puffin_manager/writer.rs

```diff
@@ -208,3 +163,40 @@ where
         Ok(size)
     }
 }
+
+impl<B, G, W> CachedPuffinWriter<B, G, W>
+where
+    B: BlobGuard,
+    G: DirGuard,
+    W: AsyncWrite + Unpin + Send,
+{
+    /// Compresses the raw data and writes it to the puffin file.
+    async fn handle_compress(
+        &mut self,
+        key: String,
+        raw_data: impl AsyncRead + Send,
+        compression: Option<CompressionCodec>,
+    ) -> Result<u64> {
+        match compression {
+            Some(CompressionCodec::Lz4) => UnsupportedCompressionSnafu { codec: "lz4" }.fail(),
```

Comment:
Not entirely sure why we are adding an unsupported compression. If it is not supported yet, I believe we can simply remove it until it is implemented.

Reply:
The puffin format supports lz4. The implementation here supports deserialization, but does not support reading the corresponding blob.